test/test_unistd.rs

```diff
@@ -457,3 +457,82 @@ fn test_canceling_alarm() {
     assert_eq!(alarm::set(60), None);
     assert_eq!(alarm::cancel(), Some(60));
 }
+
+#[test]
+fn test_getpwnam() {
+    let res = User::query( UserQuery::Name("root".to_string()) ).unwrap();
+    assert!(res.unwrap().uid == Uid::from_raw(0));
+}
+
+#[test]
+fn test_getpwuid() {
+    let res = User::query( UserQuery::Uid(Uid::from_raw(0)) ).unwrap();
+    assert!(res.unwrap().uid == Uid::from_raw(0));
+}
+
+#[test]
+fn test_getgrgid() {
+    let res = Group::query( GroupQuery::Gid(Gid::from_raw(0)) ).unwrap();
+    assert!(res.unwrap().gid == Gid::from_raw(0));
+}
+
+#[test]
+#[cfg(not(any(target_os = "android", 
+              target_os = "ios", 
+              target_os = "macos", 
+              target_env = "musl",
+              target_arch = "mips",
+              target_arch = "mips64",
+              target_arch = "mips64el",
+              target_arch = "mipsel",
+              target_arch = "s390x")))] 
+fn test_users_iterator() {
+    #[allow(unused_variables)]
+    let m = ::USER_GRP_ITER_MTX.lock().expect("Mutex got poisoned by another test");
```

Comment:
Change the name to `_m` and you won't need the `#[allow(unused_variables)]`

Reply:
I'm pretty sure the reason that other tests do it this way is because they were added back when the minimum compiler version was 1.14.0 or something, and it didn't understand that leading underscores denoted unused variables.

Reply:
I think that would be best left to another PR, since I'd be editing many other unrelated tests, which I'd be happy to work on after this one.

Reply:
Let's add the underscore and get rid of the `#[allow(` as part of this PR. You don't need to clean up existing examples, but since this is new code it should do this the right way from the start.